src/vmm/src/vstate/vcpu.rs

```diff
@@ -486,6 +510,31 @@ impl Vcpu {
         StateMachine::finish()
     }
 
+    fn handle_userfault(
+        &mut self,
+        userfaultfd_data: UserfaultData,
+    ) -> Result<VcpuEmulation, VcpuError> {
+        self.response_sender
+            .send(VcpuResponse::Userfault(userfaultfd_data))
+            .expect("Failed to send userfault data");
+        self.exit_evt.write(1).expect("Failed to write exit event");
+
+        if let Some(ref userfault_resolved) = self.userfault_resolved {
+            let (lock, cvar) = &**userfault_resolved;
+            let mut val = lock
+                .lock()
+                .expect("Failed to lock userfault resolved mutex");
+            while !*val {
+                val = cvar.wait(val).unwrap();
+            }
+            *val = false;
+        } else {
+            panic!("Vcpu::handle_userfault called without userfault_resolved condvar");
```

Comment:
Instead of `if let` and manually reborrowing, we can just do 

```rs
let (lock, cvar) = self.userfault_resolved.as_deref().expect("Vcpu::handler_userfault called without userfault_resolved condvar");
```